pino.js

```diff
@@ -160,23 +172,33 @@ Object.defineProperty(pinoPrototype, 'child', {
     }
     data = this.chindings + data.substr(0, data.length - 1)
 
+    let levelChild = bindings.level || this.level
+    const level = this.levelMap
+    Object.keys(level).sort(tools.byPrecision).map((pattern) => {
+      const regexp = new RegExp('^' + pattern.replace(/\*/g, '.*?') + '$')
+      if (bindings.name && regexp.test(bindings.name)) levelChild = level[pattern]
+    })
```

Comment:
thiw should be a for loop, otherwise we are losing performance on child creation.